content/docs/1.6.0/advanced-resources/os-distro-specific/additional-configurations-on-okd.md

```diff
@@ -0,0 +1,82 @@
+---
+title:  Additional Configurations on OCP/OKD
+weight: 4
+---
+
+To operate Longhorn on a cluster provisioned with OpenShift 4.x, some additional configuration is required.
+
+## Default Path /var/lib/longhorn Setup
+
+Label each node for storage with:
+
+```bash
+oc get nodes --no-headers | awk '{print $1}'
+oc label node "${NODE_NAME}" node.longhorn.io/create-default-disk=true
+```
+
+## Separate Path /var/mnt/longhorn Setup
+
+### Create Filesystem for devices
+
+On the storage nodes create a filesystem with the label longhorn:
+
+```bash
+oc get nodes --no-headers | awk '{print $1}'
+oc debug node/${NODE_NAME} -t -- chroot /host bash
+```
+
+Check devices are present and format them with Longhorn label:
+
+```bash
+lsblk
+sudo mkfs.ext4 -L longhorn /dev/${DEVICE_NAME}
+```
+
+### Mounting Devices On Boot with MachineConfig
+
+The Secondary Drive needs to be mounted on every boot. Save the concents and apply the `MachineConfig` with `oc apply -f`:
+
+```yaml
+apiVersion: machineconfiguration.openshift.io/v1
+kind: MachineConfig
+metadata:
+  labels:
+    machineconfiguration.openshift.io/role: worker
+  name: 71-mount-storage-worker
+spec:
+  config:
+    ignition:
+      version: 3.2.0
+    systemd:
+      units:
+        - name: var-mnt-longhorn.mount
+          enabled: true
+          contents: |
+            [Unit]
+            Before=local-fs.target
+            [Mount]
+            Where=/var/mnt/longhorn
+            What=/dev/disk/by-label/longhorn
+            Options=rw,relatime,discard
+            [Install]
+            WantedBy=local-fs.target
+```
+
+### Label and Annotate Nodes
+
+Label and annotate storage nodes by:
+
+```bash
+oc get nodes --no-headers | awk '{print $1}'
+
+oc annotate node ${NODE_NAME} --overwrite node.longhorn.io/default-disks-config='[{"path":"/var/mnt/longhorn","allowScheduling":true}]'
+oc label node ${NODE_NAME} --overwrite node.longhorn.io/create-default-disk=config
+```
+
+## Install Longhorn
```

Comment:
Do we need to add Uninstallation or Upgrade here?

Reply:
Added. Thanks